NightscoutServiceKit/Extensions/DeviceStatus.swift

```diff
@@ -0,0 +1,128 @@
+            let values = predictedGlucose.map { $0.quantity }
+            predicted = PredictedBG(startDate: startDate, values: values)
+        }
+
+        var recommended: RecommendedTempBasal?
+        if let tempBasalRecommendationDate = storedDosingDecision.tempBasalRecommendationDate {
+            recommended = RecommendedTempBasal(timestamp: tempBasalRecommendationDate.date, rate: tempBasalRecommendationDate.recommendation.unitsPerHour, duration: tempBasalRecommendationDate.recommendation.duration)
+        }
+
+        var loopEnacted: LoopEnacted?
+        if case .some(.tempBasal(let tempBasal)) = storedDosingDecision.pumpManagerStatus?.basalDeliveryState /*, lastTempBasalUploaded?.startDate != tempBasal.startDate  TODO: How to address this? */ {
+            let duration = tempBasal.endDate.timeIntervalSince(tempBasal.startDate)
+            loopEnacted = LoopEnacted(rate: tempBasal.unitsPerHour, duration: duration, timestamp: tempBasal.startDate, received:
+                true)
+//            lastTempBasalUploaded = tempBasal // TODO: How to address this?
+        }
+
+        let loopName = Bundle.main.bundleDisplayName
+        let loopVersion = Bundle.main.shortVersionString
+
+        //this is the only pill that has the option to modify the text
+        //to do that pass a different name value instead of loopName
+        let loopStatus = LoopStatus(name: loopName, version: loopVersion, timestamp: storedDosingDecision.date, iob: iob, cob: cob, predicted: predicted, recommendedTempBasal: recommended, recommendedBolus: storedDosingDecision.recommendedBolus, enacted: loopEnacted, failureReason: storedDosingDecision.error)
+
+        let pumpStatus: PumpStatus?
+
+        if let pumpManagerStatus = storedDosingDecision.pumpManagerStatus {
+
+            let battery: BatteryStatus?
+
+            if let chargeRemaining = pumpManagerStatus.pumpBatteryChargeRemaining {
+                battery = BatteryStatus(percent: Int(round(chargeRemaining * 100)), voltage: nil, status: nil)
+            } else {
+                battery = nil
+            }
+
+            let bolusing: Bool
+            if case .inProgress = pumpManagerStatus.bolusState {
+                bolusing = true
+            } else {
+                bolusing = false
+            }
+
+            let currentReservoirUnits: Double?
+            if let lastReservoirValue = storedDosingDecision.lastReservoirValue, lastReservoirValue.startDate > Date().addingTimeInterval(.minutes(-15)) {
+                currentReservoirUnits = lastReservoirValue.unitVolume
+            } else {
+                currentReservoirUnits = nil
+            }
+
+            pumpStatus = PumpStatus(
+                clock: storedDosingDecision.date,
+                pumpID: pumpManagerStatus.device.localIdentifier ?? "Unknown",
+                manufacturer: pumpManagerStatus.device.manufacturer,
+                model: pumpManagerStatus.device.model,
+                iob: nil,
```

Comment:
Why is `iob` `nil`? Is it that this value should be reported from the pump?

Reply:
This code (and much of the surrounding code) was simply moved/copied and slightly adapted from `NightscoutService.swift` in this PR. Going to leave as-is assuming the original author intended this.